components/suggest/src/store.rs

```diff
@@ -389,124 +384,115 @@ where
             DEFAULT_RECORDS_TYPES.to_vec()
         };
 
+        let fetch_fakespot_records = ingest_record_types
+            .iter()
+            .any(|r| *r == SuggestRecordType::Fakespot);
+        let records = self
+            .settings_client
+            .get_records(writer, fetch_fakespot_records)?;
```

Comment:
What's your thought on cache corruption handling? Let's say if the db is corrupted and `get_records_for_collection()` always fails and returns early from `let cache = db.read(|dao| dao.read_cached_rs_data(collection_name));`. It will never try to fetch records from RS due to the absence of the cache, correct?

Reply:
We should definitely be handling this differently.  I changed it so the main methods never fail, they just log warnings and return `None`.  I think that's the best way to handle things, it was definitely nice that ingestion continued to work when I switched the format from JSON to msgpack.